### PR TITLE
Feature/fpp links rendering fix

### PR DESCRIPTION
**Pull request by CompuCell3D**
Please checkout branch on your machine and run few simulations with FPP plugin to make sure the behavior is reasonable

This is PR addresses several issues

1) it improves rendering of fpp links
2) it changes how we compute distance in the fpp plugin to obey boundary conditions. Previous code was not correct in this regard. However, when the length of the link was much smaller than "avg lattice size" (as it was in most simulations) this did not create an issue. The issue occurred when e.g. you have link in the z direction and its length is > 0.5 of the z dimension.
3) Changes to invariant_distance/invariant_distance_vector to respect boundary conditions. Same motivation , problems, and impact as in 2)
 
Link rendering can still be improved but overall this version is much better and than what we had before

Issues addressed in this PR:
https://github.com/CompuCell3D/CompuCell3D/issues/350
https://github.com/CompuCell3D/CompuCell3D/issues/347
https://github.com/CompuCell3D/CompuCell3D/issues/312

### CODE REVIEW

File: CompuCell3D/core/CompuCell3D/plugins/FocalPointPlasticity/FocalPointPlasticityPlugin.cpp [hunk 1]
```diff
@@ -250,76 +231,30 @@ void FocalPointPlasticityPlugin::update(CC3DXMLElement *_xmlData, bool _fullInit
...
 	}
 
 	plastParamsArray.clear();
 	plastParamsArray.assign(size,vector<FocalPointPlasticityTrackerData>(size,FocalPointPlasticityTrackerData()));
 
 	for(int i = 0 ; i < cellTypesVector.size() ; ++i)
 		for(int j = 0 ; j < cellTypesVector.size() ; ++j){
-			//cerr<<"cellTypesVector[i]="<<(int)cellTypesVector[i]<<endl;
-			//cerr<<"cellTypesVector[j]="<<(int)cellTypesVector[j]<<endl;
+			cerr<<"cellTypesVector[i]="<<(int)cellTypesVector[i]<<endl;
```
Comment (JulianoGianlupi) [inline CompuCell3D/core/CompuCell3D/plugins/FocalPointPlasticity/FocalPointPlasticityPlugin.cpp:250]: Do we want to keep the prints?

**Review by JulianoGianlupi — APPROVED** [review]

File: CompuCell3D/core/CompuCell3D/plugins/FocalPointPlasticity/FocalPointPlasticityPlugin.cpp (hunk 1 above)
Comment (CompuCell3D) [reply CompuCell3D/core/CompuCell3D/plugins/FocalPointPlasticity/FocalPointPlasticityPlugin.cpp]: good point. Fixed

File: CompuCell3D/core/CompuCell3D/Boundary/BoundaryStrategy.cpp [hunk 2]
```diff
@@ -84,9 +86,17 @@ BoundaryStrategy::BoundaryStrategy(string boundary_x, string boundary_y,
...
 
+    boundaryConditionIndicator.assign(3, 0);
+
 	strategy_x = BoundaryFactory::createBoundary(boundary_x);
 	strategy_y = BoundaryFactory::createBoundary(boundary_y);
 	strategy_z = BoundaryFactory::createBoundary(boundary_z);
+
+    boundaryConditionIndicator[0] = boundary_x == "Periodic" ? 1 : 0;
+    boundaryConditionIndicator[1] = boundary_y == "Periodic" ? 1 : 0;
+    boundaryConditionIndicator[2] = boundary_z == "Periodic" ? 1 : 0;
```
Comment (tjsego) [inline CompuCell3D/core/CompuCell3D/Boundary/BoundaryStrategy.cpp:97]: Would it be better to declare labels for all boundary condition types and assign them here instead?

**Review by tjsego — COMMENTED** [review]
Just one comment to address. Otherwise this looks good.

File: CompuCell3D/core/CompuCell3D/Boundary/BoundaryStrategy.cpp (hunk 2 above)
Comment (CompuCell3D) [reply CompuCell3D/core/CompuCell3D/Boundary/BoundaryStrategy.cpp:97]: Agree, we can make a bigger PR to fix hardcoded strings

Comment (CompuCell3D) [reply CompuCell3D/core/CompuCell3D/Boundary/BoundaryStrategy.cpp:97]: added new issue to address it https://github.com/CompuCell3D/CompuCell3D/issues/358

**Pull request merged**